modexp/src/lib.rs

```diff
@@ -0,0 +1,335 @@
+extern crate ewasm_api;
+extern crate num_bigint;
+
+use num::{One, Zero};
+use num_bigint::{BigInt, Sign};
+use std::cmp;
+use std::ops::{Rem, Shl};
+
+fn f(x: usize) -> u64 {
+    let cost = match x {
+        0...64 => x * x,
+        65...1024 => (x * x / 4) + 96 * x - 3072,
+        _ => (x * x / 16) + 480 * x - 199680,
+    };
+    cost as u64
```

Comment:
I think this expression can be shortened, although it is personal preference.
Something like 
```
match x {
...
} as u64
```